include/uk/arch/types.h

```diff
@@ -251,6 +251,8 @@ typedef __u64 __paddr_t;
 #define __NULL ((void *) 0)
 #endif
 
+typedef char *caddr_t; /* core address */
```

Comment:
Is this `typedef` really needed? I see we only have [one instance](https://github.com/unikraft/unikraft/blob/staging/lib/ukring/ring.c#L49) where it is used and I'd rather have it changed there with `char *` than have a completely new type defined.

Reply:
This type is used in ukring and was added in commit: 7f212b4df81d54fca4b647e09f2c52082875f707 and I've met a compilation error due to the missing definition. I assumed this type to be architecture dependent that's why I've added type definition in arch/types.h, rather then changed ukring library with unknown (for me at least) side effects on other platforms.